rfcs/001-libp2p-resolver.md

```diff
@@ -0,0 +1,82 @@
+This protocol allows peers to exchange metadata about themselves upon connection. This metadata includes:
+
+- [Peer ID][5]: A unique identifier for the peer.
+- [Multiaddrs][4]: Addresses where the peer can be reached.
+- Supported Protocols: A list of protocols the peer supports.
+
+Peer discovery is done using the [rendezvous protocol][3], this protocol involves a system where peers join a shared namespace to advertise themselves and find others.
+When a new peer wants to participate in the network, it first connects to a bootstrap node (also known as a rendezvous point), which is a well-known and reachable peer that helps initialize connections within the network.
+Once connected to the bootstrap node, the new peer uses the [rendezvous protocol][3] to discover other peers. The [rendezvous protocol][3] operates by having peers register their presence under a specific namespace with a designated rendezvous point.
+Peers seeking connections to others interested in the same namespace query the rendezvous point, which returns a list of peer addresses registered under that namespace.
+
+## Multi-Address Resolution
+
+Once the target peer is discovered, the application will use the [identify protocol][3] to retrieve detailed information, including the peer's [multi-addresses][4], supported protocols, and additional metadata.
+
+The [identify protocol][3] facilitates the exchange of information between peers once a connection is established. After discovering the target peer via the [rendezvous protocol][2], the application will:
+
+- Establish a direct connection to the target peer using its discovered address.
+- Initiate the [identify protocol][3] handshake to request identity information.
+- Extract and display the [multi-addresses][4] and associated data provided by the target peer.
+
+The application will output this information in a human-readable format, making it easy for clients to consume and utilize in further operations.
+The combination of the [rendezvous protocol][2] and [identify protocol][3] ensures a robust and modular approach to both discovering and resolving peers in the network.
+
+## Command-Line Interface
+
+The application will provide an intuitive CLI interface, allowing clients to input a [Peer ID][5] and retrieve its [multi-addresses][4].
+Additional options will enable customization of settings, such as specifying bootstrap nodes or adjusting discovery parameters.
+The output will include a list of [multi-addresses][4] associated with the [Peer ID][5], presented in a structured and readable format.
+
+## Configuration and Deployment
+
+The application will allow configuration through a settings file or environment variables.
+Clients can define bootstrap nodes, set timeouts for peer discovery and identify protocol exchanges, and specify logging preferences.
+The CLI tool will be distributed as a single executable binary, ensuring ease of deployment across diverse platforms and environments.
```

Comment:
I don't think this is relevant, IMO the RFC is to discuss the high-level ideas; CLI, single or multiple binaries, etc, those are implementation details.

https://github.com/eigerco/polka-storage/pull/629/files#diff-96ebdac0b36633969fe397b19ef41c994f26b798a7bff824a8ca1f5486aa0f7eR22

This line can almost be thought of as the border, it states that the final product should be configurable, provides and example, but it doesn't really commit to anything